numpy/core/src/multiarray/buffer.c

```diff
@@ -248,14 +260,17 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
             /* Insert padding manually */
             if (*offset > new_offset) {
                 PyErr_SetString(PyExc_RuntimeError,
-                                "This should never happen: Invalid offset in "
-                                "buffer format string generation. Please "
-                                "report a bug to the Numpy developers.");
+                                "The buffer interface does not support "
+                                "overlapping fields");
```

Comment:
What are we going to do about #6053 here? Seems like it's going to make casting to memoryview a lot harder...